aws/request/request.go

```diff
@@ -114,6 +114,40 @@ func New(cfg aws.Config, clientInfo metadata.ClientInfo, handlers Handlers,
 // using a WithContext API operation method.
 type Option func(*Request)
 
+// WithGetResponseHeader builds a request Option which will retrieve a single
+// header value from the HTTP Response. If there are multiple values for the
+// header key use WithGetResponseHeaders instead to access the http.Header
+// map directly. The passed in val pointer must be non-nil.
+//
+// This Option can be used multiple times with a single API operation.
+//
+//    var id2, versionID string
+//    svc.PutObjectWithContext(ctx, params,
+//        request.WithGetResponseHeader("x-amz-id-2", &id2),
+//        request.WithGetResponseHeader("x-amz-version-id", &versionID),
+//    )
+func WithGetResponseHeader(key string, val *string) Option {
+	return func(r *Request) {
+		r.Handlers.Complete.PushBack(func(req *Request) {
```

Comment:
A user could pass in a nil value here especially if (s)he is taking it from a shape. May need to document that this will panic if `nil` is passed.